File: test/fixtures/csharp/upload_file.cs
using System.Net.Http;

HttpClient client = new HttpClient();

HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Put, "http://localhost:28139/file.txt");
request.Content = new ByteArrayContent(File.ReadAllBytes("file.txt"));

HttpResponseMessage response = await client.SendAsync(request);
response.EnsureSuccessStatusCode();
string responseBody = await response.Content.ReadAsStringAsync();

File: test/fixtures/java/upload_file.java
import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpRequest.BodyPublishers;
import java.net.http.HttpResponse;
import java.nio.file.Paths;

HttpClient client = HttpClient.newHttpClient();

HttpRequest request = HttpRequest.newBuilder()
    .uri(URI.create("http://localhost:28139/file.txt"))
    .PUT(BodyPublishers.ofFile(Paths.get("file.txt")))
    .build();

HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
